### PR TITLE
Start Vitess automatically when we run tests

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/PingDatabaseService.kt [hunk 1]
```diff
@@ -31,10 +31,14 @@ class PingDatabaseService @Inject constructor(
...
-            c.createStatement().executeQuery("SELECT 1 FROM dual").uniqueResult { it.getInt(1) }
-        check(result == 1)
+      try {
+        dataSource.connection.use { c ->
+          val result =
+              c.createStatement().executeQuery("SELECT 1 FROM dual").uniqueResult { it.getInt(1) }
+          check(result == 1)
+        }
+      } catch (e: Exception) {
+        e.printStackTrace()
```
Comment (eeSeeGee) [inline misk-hibernate/src/main/kotlin/misk/hibernate/PingDatabaseService.kt]: You don't want to throw here?

File: .travis.yml [hunk 2]
```diff
@@ -4,12 +4,6 @@ jdk:
   - oraclejdk8
   - oraclejdk9
 
-before_script:
-  - docker pull vitess/base
-  - pushd misk-hibernate/src/test/resources/misk/hibernate
-  - docker-compose up -d
```
Comment (drashkov) [inline .travis.yml]: You can also do docker-compose -f <file> so you don't need to change dirs.

File: misk-hibernate-testing/src/main/kotlin/misk/hibernate/StartVitessService.kt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    docker.startContainerCmd(containerId!!).exec()
+    StartVitessService.logger.info("Started Vitess with container id $containerId")
+  }
+
+  fun stop() {
+    if (!isRunning()) {
+      return
+    }
+
+    docker.killContainerCmd(containerId!!).exec()
```
Comment (eeSeeGee) [inline misk-hibernate-testing/src/main/kotlin/misk/hibernate/StartVitessService.kt]: Does the container also need to be `rm`'d?

File: misk-hibernate/src/main/kotlin/misk/hibernate/PingDatabaseService.kt (hunk 1 above)
Comment (tirsen) [reply misk-hibernate/src/main/kotlin/misk/hibernate/PingDatabaseService.kt]: Oops me debugging

Comment (tirsen) [reply misk-hibernate/src/main/kotlin/misk/hibernate/PingDatabaseService.kt]: reverting

File: misk-hibernate-testing/src/main/kotlin/misk/hibernate/StartVitessService.kt (hunk 3 above)
Comment (tirsen) [reply misk-hibernate-testing/src/main/kotlin/misk/hibernate/StartVitessService.kt]: Hmm I thought `kill` deleted it too. The container isn't there when I check with `docker container ls`.

Comment (eeSeeGee) [reply misk-hibernate-testing/src/main/kotlin/misk/hibernate/StartVitessService.kt]: Ah, yeah I think that's right.

**Pull request merged**